### PR TITLE
Fix/rs

### PR DESCRIPTION
**Pull request by Raycoms**
Fix request handling

# Changes proposed in this pull request:
- Fixes async requests
- Fixes retrying resolver
- Fix stuck cook
- turn off teleport to upper levels and fix horde issues
- avoid barbarians correctly and clean horde list
- fix barbarian close calc

Review please

### CODE REVIEW

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New  | Old  | Diff |
| ------ |:-----:|:----:|:----:|:----:|
| Total  | 34017 | 3655 | 2722 | +933 |
| Errors |  593  |  26  |  17  |  +9  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=11896&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 34017 |  0  |  0  |  0   |
| Errors |  593  |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 34011 |  7  | 13  |  -6  |
| Errors |  593  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=12116&tab=Inspection&buildTypeId=bt88)

File: src/api/java/com/minecolonies/api/colony/requestsystem/requestable/Stack.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public boolean matches(@NotNull final ItemStack stack)
     {
         if (matchOreDic)
         {
-            return OreDictionary.itemMatches(getStack(), stack, matchMeta) && getCount() <= stack.getCount();
+            return OreDictionary.itemMatches(getStack(), stack, matchMeta);
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/colony/requestsystem/requestable/Stack.java:133]: Please revert this Change. This contraticts the other behaviour of Stack.

**Review by marchermans — CHANGES_REQUESTED** [review]
One small Thing.

**Pull request merged**